btx/interfaces/ipsana.py

```diff
@@ -109,12 +112,11 @@ def get_wavelength_evt(self, evt):
         wavelength : float
             wavelength in Angstrom
         """
-        photon_energy = self.get_photon_energy_eV_evt(evt)
-        if np.isinf(photon_energy) or photon_energy is None:
-            return self.get_wavelength()
-        else:
-            lambda_m =  1.23984197386209e-06 / photon_energy # convert to meters using e=hc/lambda
+        try:
+            lambda_m = 1.23984197386209e-06 / self.get_photon_energy_eV_evt(evt) # convert to meters using e=hc/lambda
```

Comment:
I think it might be better to handle this with a conditional.

Python short-circuits conditional evaluation so reversing the order of the comparisons in the conditional of `get_wavelength_evt` should eliminate the `TypeError`:
```
        if photon_energy is None or np.isinf(photon_energy):
            return self.get_wavelength()
        else:
            lambda_m =  1.23984197386209e-06 / photon_energy
```

Tested on run 131 of mfpx23120, and it seems to work, but maybe you can confirm.

Reply:
OK done